freud/plot.py

```diff
@@ -0,0 +1,272 @@
+        pmft_arr[np.isinf(pmft_arr)] = np.nan
+        dx = (2.0 * 3.0) / pmft.n_bins_X
+        dy = (2.0 * 3.0) / pmft.n_bins_Y
+        nan_arr = np.where(np.isnan(pmft_arr))
+        for i in range(pmft.n_bins_X):
+            x = -3.0 + dx * i
+            for j in range(pmft.n_bins_Y):
+                y = -3.0 + dy * j
+                if ((x*x + y*y < 1.5) and (np.isnan(pmft_arr[j, i]))):
+                    pmft_arr[j, i] = 10.0
+        w = int(2.0 * pmft.n_bins_X / (2.0 * 3.0))
+        center = int(pmft.n_bins_X / 2)
+
+        # Get the center of the histogram bins
+        pmft_smooth = gaussian_filter(pmft_arr, 1)
+        pmft_image = np.copy(pmft_smooth)
+        pmft_image[nan_arr] = np.nan
+        pmft_smooth = pmft_smooth[center-w:center+w, center-w:center+w]
+        pmft_image = pmft_image[center-w:center+w, center-w:center+w]
+        x = pmft.X
+        y = pmft.Y
+        reduced_x = x[center-w:center+w]
+        reduced_y = y[center-w:center+w]
+
+        # Plot figures
+        fig = Figure(figsize=(12, 5), facecolor='white')
+        values = [-2, -1, 0, 2]
+        norm = Normalize(vmin=-2.5, vmax=3.0)
+        n_values = [norm(i) for i in values]
+        colors = viridis(n_values)
+        colors = colors[:, :3]
+        # verts = make_polygon(sides=6, radius=0.6204)
+        lims = (-2, 2)
+        ax0 = fig.add_subplot(1, 2, 1)
+        ax1 = fig.add_subplot(1, 2, 2)
+        for ax in (ax0, ax1):
+            # commented out since in general, particles are not hexagonal
+            # ax.contour(reduced_x, reduced_y, pmft_smooth,
+            #            [9, 10], colors='black')
+            # ax.contourf(reduced_x, reduced_y, pmft_smooth,
+            #             [9, 10], hatches='X', colors='none')
+            # ax.plot(verts[:,0], verts[:,1], color='black', marker=',')
+            # ax.fill(verts[:,0], verts[:,1], color='black')
+            ax.set_aspect('equal')
+            ax.set_xlim(lims)
+            ax.set_ylim(lims)
+            ax.xaxis.set_ticks([i for i in range(lims[0], lims[1]+1)])
+            ax.yaxis.set_ticks([i for i in range(lims[0], lims[1]+1)])
+            ax.set_xlabel(r'$x$')
+            ax.set_ylabel(r'$y$')
+
+        ax0.set_title('PMFT Heat Map')
+        im = ax0.imshow(np.flipud(pmft_image),
+                        extent=[lims[0], lims[1], lims[0], lims[1]],
+                        interpolation='nearest', cmap='viridis',
+                        vmin=-2.5, vmax=3.0)
+        ax1.set_title('PMFT Contour Plot')
+        ax1.contour(reduced_x, reduced_y, pmft_smooth,
+                    [-2, -1, 0, 2], colors=colors)
+
+        fig.subplots_adjust(right=0.85)
+        cbar_ax = fig.add_axes([0.88, 0.1, 0.02, 0.8])
+        fig.colorbar(im, cax=cbar_ax)
+        f = io.BytesIO()
+        # Sets an Agg backend so this figure can be rendered
+        FigureCanvasAgg(fig)
+        fig.savefig(f, format='png')
+        return f.getvalue()
```

Comment:
A lot of this plotting logic is specific to the specific data in the PMFT example. I would simplify this to the bare minimum required to generate a simple PMFT plot without the bells and whistles to make the plot especially pretty. My inclination is to just do the simple heat map and remove the extra logic for the contour as well. @bdice may disagree with some of these recommendations though.

Reply:
Yes, I agree we should simplify this to the minimum.

Reply:
<img width="562" alt="Screen Shot 2019-06-13 at 9 40 19 PM" src="https://user-images.githubusercontent.com/15791520/59477633-1394da80-8e24-11e9-9606-2f9b2c8d0e8b.png">

Something like this?
Or is this still specific to the data?

Reply:
@jinsoo960 The code is doing a bunch of scaling and trickery that is specific to this hexagon analysis - the colorbar needs to find its min/max bounds from the min/max of the PMFT, and the x/y limits of the plot should correspond to the PMFT's actual bin locations. Last I checked, it was scaling the axes by 2/3 and only showing a cropped version of the actual data, for the purposes of this plot.

Reply:
<img width="582" alt="Screen Shot 2019-06-13 at 10 46 06 PM" src="https://user-images.githubusercontent.com/15791520/59480047-1647fd80-8e2d-11e9-9122-fdec33ed2908.png">
<img width="524" alt="Screen Shot 2019-06-13 at 10 46 16 PM" src="https://user-images.githubusercontent.com/15791520/59480049-19db8480-8e2d-11e9-9e07-ff7dd2c7dec2.png">

This is an almost bare minimum version, with smoothing an all other logic removed.
Only processing done is changing `inf` to `nan`.

Reply:
Close- bring back the part that labels the `x` and `y` ticks by the values of `pmft.X` and `pmft.Y`, instead of labeling them by the index of the bin.

Reply:
Also change the title of the plot to `2D Potential of Mean Force`.